lib/query-builder-fields.js

```diff
@@ -0,0 +1,360 @@
+'use strict';
+
+const QueryBuilderError = require('./query-builder-error');
+
+class QueryBuilderSelect {
+
+	static get selectFunctions() {
+		return {
+			count: 'count',
+			min: 'min',
+			max: 'max',
+			sum: 'sum',
+			avg: 'avg'
+		};
+	}
+
+	/**
+	 * Initialize the necesary data to use the Builder
+	 * @param {instance} model Model to be used
+	 */
+	static initModel(model) {
+
+		this.modelName = model.constructor.name;
+		this.fields = model.constructor.fields;
+		this.flags = model.constructor.flags;
+		this.joins = model.constructor.joins;
+	}
+
+	/**
+	 * Build Select. Call Knex Select with fields in params
+	 * @param {object} knex must have key 'knexStatement' with knex function
+	 * @param {instance} model
+	 * @param {object} params
+	 */
+	static buildSelect(knex, model, params) {
+
+		if(!knex)
+			throw new QueryBuilderError('Invalid Knex', QueryBuilderError.codes.INVALID_KNEX);
+
+		if(!model)
+			throw new QueryBuilderError('Invalid Model', QueryBuilderError.codes.INVALID_MODEL);
+
+
+		this.initModel(model);
+
+		this.params = params;
+
+		// eslint-disable-next-line prefer-destructuring
+		const fields = params.fields;
+
+		let fieldsForSelect;
+		let flagFieldsForSelect = false;
+
+		if(typeof fields === 'undefined') {
+			fieldsForSelect = 't.*';
+
+			if(typeof this.params.noFlags === 'undefined' || !this.params.noFlags)
```

Comment:
```suggestion
			if(!this.params.noFlags)
```
Si le agregamos `!this.params.noFlags` al `type of === undefined`, estamos haciendo las mismas comprobaciones al final.